bin/all-autosens-history.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+oref0_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+interval=${1:-96}
+allowedload=${2:-5}
+
+main() {
+    ls | while read participant; do
+        (
+            echo "Processing participant $participant"
+            cd $participant/direct-sharing-31/ || die "$participant/direct-sharing-31/ not found"
+            gunzip *.gz 2>/dev/null
+            mkdir -p parts || die "Couldn't mkdir parts"
+            rm parts/treatments*.json 2>/dev/null
+            echo Checking / waiting for system load to be below $allowedload before continuing
+            while highload; do
+                sleep 30
+            done
+            # this assumes that there are always fewer treatment records than entries, so the treatments will finish first
+            # (or at least finish before the oref0-autosens-history.js job for that month starts)
+            cat treatments*.json | jq -cn --stream 'fromstream(1|truncate_stream(inputs))' | while read line; do
+                date=$(echo $line | jq .created_at)
+                year=$(echo $date | cut -b 2,3,4,5)
+                month=$(echo $date | cut -b 7,8)
+                echo "Processing participant $participant treatments for $year $month"
+                echo $line >> parts/treatments-$year-$month.json
+            done | uniq &
+            rm parts/entries*.json
+            cat entries*.json | jq -cn --stream 'fromstream(1|truncate_stream(inputs))' | while read line; do
+                date=$(echo $line | jq .dateString)
+                year=$(echo $date | cut -b 2,3,4,5)
+                month=$(echo $date | cut -b 7,8)
+                if [[ $year =~ 201. ]]; then
+                    echo "Processing participant $participant entries for $year $month"
+                    echo $line >> parts/entries-$year-$month.json
+                fi
+            done | uniq
+            for year in 2018 2017 2016; do
```

Comment:
This is missing a value now ;)
```suggestion
            for year in 2019 2018 2017 2016; do
```